### PR TITLE
SFR-2311: API Edge Test Cases

### PR DESCRIPTION
**Pull request by ayan1229**
This PR adds edge cases for following API end points

- /edition/{id}
- /link/{id}
- /search
- /work/{uuid}

### CODE REVIEW

**Review by kylevillegas93 — CHANGES_REQUESTED** [review]
These changes look good! I recommend using parametrize to reduce the number of tests, e.g. 

```
import pytest

@pytest.mark.parametrize("test_id, expected_status", [
    ('00000000-0000-0000-0000-000000000000', 400), 
    ('invalid_id_format', 400),
    ('', 404),
    ('%$@!*', 400),
])
def test_get_edition_invalid_ids(test_id, expected_status):
    url = API_URL + f'/edition/{test_id}'
    
    response = requests.get(url)

    assert_response_status(url, response, expected_status)
```

File: tests/integration/api/test_search_collection.py [hunk 1]
```diff
@@ -13,3 +13,23 @@ def test_get_collection():
...
+
+def test_get_collection_empty_id():
+    url = API_URL + '/collection/'
+    response = requests.get(url)
+    assert_response_status(url, response, 404)
+
+def test_get_collection_special_characters():
+    url = API_URL + '/collection/%$@!*'
+    response = requests.get(url)
+    assert_response_status(url, response, 400)
```
Comment (kylevillegas93) [inline tests/integration/api/test_search_collection.py:35]: nit: let's add a new line at the end of the file

File: tests/integration/api/test_get_edition.py [hunk 2]
```diff
@@ -14,3 +14,23 @@ def test_get_collection():
...
+def test_get_edition_empty_id():
+    url = API_URL + '/edition/'
+    response = requests.get(url)
+    assert_response_status(url, response, 404)
+
+def test_get_edition_special_characters():
+    url = API_URL + '/edition/%$@!*'
+    response = requests.get(url)
+    assert_response_status(url, response, 400)
+
```
Comment (kylevillegas93) [inline tests/integration/api/test_get_edition.py:36]: same this should just be a new line - no other white space

File: tests/integration/api/test_search_collection.py
Comment (kylevillegas93) [inline tests/integration/api/test_search_collection.py:1]: let's separate the search and collection tests into separate files

Comment (kylevillegas93) [inline tests/integration/api/test_search_collection.py:1]: let's just update this file to `test_search.py`

**Review by kylevillegas93 — APPROVED** [review]
Looks great - let's just rename one file!
